test/applications_integration_test.go

```diff
@@ -757,11 +808,153 @@ func checkRandomElementResult(resultIndex int, input string) error {
 	return nil
 }
 
-//@applications.verified
+func theContentsOfTheBoxWithNameShouldBeIfThereIsAnErrorItIs(fromClient, encodedBoxName, boxContents, errStr string) error {
+	var box models.Box
+	var err error
+	if fromClient == "algod" {
+		box, err = algodV2client.GetApplicationBoxByName(applicationId).Name(encodedBoxName).Do(context.Background())
+	} else if fromClient == "indexer" {
+		box, err = indexerV2client.LookupApplicationBoxByIDandName(applicationId).Name(encodedBoxName).Do(context.Background())
+	} else {
+		err = fmt.Errorf("expecting algod or indexer, got " + fromClient)
+	}
+	if err != nil {
+		if strings.Contains(err.Error(), errStr) {
+			return nil
+		}
+		return err
+	}
+
+	b64Value := base64.StdEncoding.EncodeToString(box.Value)
+	if b64Value != boxContents {
+		return fmt.Errorf("expected box value %s is not equal to actual box value %s", boxContents, b64Value)
+	}
+
+	return nil
+}
+
+func bytesContains(elem []byte, xs [][]byte) bool {
+	for _, x := range xs {
+		if bytes.Equal(elem, x) {
+			return true
+		}
+	}
+	return false
+}
+
+func currentApplicationShouldHaveFollowingBoxes(fromClient, encodedBoxesRaw string) error {
+	var expectedNames [][]byte
+	if len(encodedBoxesRaw) > 0 {
+		encodedBoxes := strings.Split(encodedBoxesRaw, ":")
+		expectedNames = make([][]byte, len(encodedBoxes))
+		for i, b := range encodedBoxes {
+			expected, err := base64.StdEncoding.DecodeString(b)
+			if err != nil {
+				return err
+			}
+			expectedNames[i] = expected
+		}
+	}
+
+	var r models.BoxesResponse
+	var err error
+
+	if fromClient == "algod" {
+		r, err = algodV2client.GetApplicationBoxes(applicationId).Do(context.Background())
+	} else if fromClient == "indexer" {
+		r, err = indexerV2client.SearchForApplicationBoxes(applicationId).Do(context.Background())
+	} else {
+		err = fmt.Errorf("expecting algod or indexer, got " + fromClient)
+	}
+	if err != nil {
+		return err
+	}
+
+	actualNames := make([][]byte, len(r.Boxes))
+	for i, b := range r.Boxes {
+		actualNames[i] = b.Name
+	}
+
+	if len(expectedNames) != len(actualNames) {
+		return fmt.Errorf("expected and actual box names length do not match:  %v != %v", len(expectedNames), len(actualNames))
+	}
+
+	for _, e := range expectedNames {
+		if !bytesContains(e, actualNames) {
```

Comment:
This test would be stronger if we directly compare the actual and expected name slices